serveradmin/serverdb/query_committer.py

```diff
@@ -616,6 +626,91 @@ def _validate_real_attributes(servertype, real_attributes):     # NOQA: C901
         violations_attribs,
     )
 
+def _resolve_servertype_changes(changed, joined_attributes):
+    """Expand servertype changes
+
+    Changing servertypes likely entails deleting attributes not available on
+    the target servertype.
+    """
+
+    # Filter for servertype changes
+    servertype_changes = [
+        change for change in changed
+        if 'servertype' in change.keys()
+    ]
+
+    # No servertype changes, we're done here
+    if not servertype_changes:
+        return changed
+
+    # Forbid deleting the servertype attribute
+    for change in servertype_changes:
+        if (
+            change['servertype']['action'] != 'update' or
+            'new' not in change['servertype'] or
+            not change['servertype']['new']
+        ):
+            CommitError('Cannot delete servertype')
+
+    # Get attributes of the new and old servertypes to check what changes
+    involved_servertypes = chain(*[
+        [change['servertype']['old'], change['servertype']['new']]
+        for change in servertype_changes
+    ])
```

Comment:
No need to `*[]`.

Reply:
Actually this doesn't seem to get flattened without `*[]` 🤔
```Internal Server Error: /servershell/commit
Traceback (most recent call last):
  File "/Users/space/.local/share/virtualenvs/serveradmin-m3Kn4UAw/lib/python3.5/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/Users/space/.local/share/virtualenvs/serveradmin-m3Kn4UAw/lib/python3.5/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/Users/space/.local/share/virtualenvs/serveradmin-m3Kn4UAw/lib/python3.5/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/Users/space/.local/share/virtualenvs/serveradmin-m3Kn4UAw/lib/python3.5/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/space/.local/share/virtualenvs/serveradmin-m3Kn4UAw/lib/python3.5/site-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "./serveradmin/servershell/views.py", line 305, in commit
    commit_query(changed=changed, deleted=deleted, user=user)
  File "./serveradmin/serverdb/query_committer.py", line 80, in commit_query
    _resolve_servertype_changes(changed, joined_attributes)
  File "./serveradmin/serverdb/query_committer.py", line 663, in _resolve_servertype_changes
    involved_servertype_ids
  File "./serveradmin/serverdb/query_committer.py", line 398, in _get_servertype_attributes
    for servertype_id in set(servertype_ids):
TypeError: unhashable type: 'list'```